modules/util.py

```diff
@@ -62,6 +62,22 @@ def getBootPartition(conffile):
         return bootdevice
     return None
 
+def getPartitionRelativeToBoot(conffile, label, relativeIndex):
+    ''' Returns the partition device path when index is relative to boot partition '''
+    # First search by label
+    partdevice = getDevice(label)
+    if not partdevice:
+        match = re.match(r"(.*?)(\d+$)", getBootPartition(conffile))
```

Comment:
That's exactly the same as partitions attribute. Do you find anything wrong with this approach? I agree it can be done with partitions list too.